src/parser/instruction.rs

```diff
@@ -42,75 +42,70 @@ pub(crate) fn parse_instruction(input: ParserInput) -> InternalParserResult<Inst
             }
-            .map_err(|err| {
-                nom::Err::Failure(
-                    InternalParseError::from_kind(
-                        &input[..1],
-                        ParserErrorKind::InvalidCommand { command: *command },
-                    )
-                    .with_previous(extract_nom_err(err)),
-                )
-            })
+            Command::GT => command::parse_comparison(ComparisonOperator::GreaterThan, remainder),
+            Command::LE => {
+                command::parse_comparison(ComparisonOperator::LessThanOrEqual, remainder)
+            }
+            Command::LT => command::parse_comparison(ComparisonOperator::LessThan, remainder),
+            Command::Fence => command::parse_fence(remainder),
+            Command::Halt => Ok((remainder, Instruction::Halt)),
+            Command::Include => command::parse_include(remainder),
+            Command::Ior => command::parse_logical_binary(BinaryOperator::Ior, remainder),
+            Command::Jump => command::parse_jump(remainder),
+            Command::JumpUnless => command::parse_jump_unless(remainder),
+            Command::JumpWhen => command::parse_jump_when(remainder),
+            Command::Label => command::parse_label(remainder),
+            Command::Load => command::parse_load(remainder),
+            Command::Measure => command::parse_measurement(remainder),
+            Command::Move => command::parse_move(remainder),
+            Command::Exchange => command::parse_exchange(remainder),
+            Command::Mul => command::parse_arithmetic(ArithmeticOperator::Multiply, remainder),
+            Command::Neg => command::parse_logical_unary(UnaryOperator::Neg, remainder),
+            Command::Nop => Ok((remainder, Instruction::Nop)),
+            Command::Not => command::parse_logical_unary(UnaryOperator::Not, remainder),
+            Command::Pragma => command::parse_pragma(remainder),
+            Command::Pulse => command::parse_pulse(remainder, true),
+            Command::RawCapture => command::parse_raw_capture(remainder, true),
+            Command::Reset => command::parse_reset(remainder),
+            Command::SetFrequency => command::parse_set_frequency(remainder),
+            Command::SetPhase => command::parse_set_phase(remainder),
+            Command::SetScale => command::parse_set_scale(remainder),
+            Command::ShiftFrequency => command::parse_shift_frequency(remainder),
+            Command::ShiftPhase => command::parse_shift_phase(remainder),
+            Command::Store => command::parse_store(remainder),
+            Command::Sub => command::parse_arithmetic(ArithmeticOperator::Subtract, remainder),
+            Command::Xor => command::parse_logical_binary(BinaryOperator::Xor, remainder),
+            other => Err(nom::Err::Failure(InternalParseError::from_kind(
+                &input[..1],
+                ParserErrorKind::UnsupportedInstruction(*other),
+            ))),
```

Comment:
This should be unreachable - if there's anything that would be caught, it should be removed from `Command` if we don't plan to support it.

Reply:
`WAIT` and `OR` have been removed. The former was discussed previously and the latter doesn't appear to even be reserved, according to the Quil spec.